pkg/hive/manager.go

```diff
@@ -160,10 +161,26 @@ func (hr *clusterManager) IsClusterDeploymentReady(ctx context.Context, doc *api
 
 func (hr *clusterManager) IsClusterInstallationComplete(ctx context.Context, doc *api.OpenShiftClusterDocument) (bool, error) {
 	cd, err := hr.hiveClientset.HiveV1().ClusterDeployments(doc.OpenShiftCluster.Properties.HiveProfile.Namespace).Get(ctx, ClusterDeploymentName, metav1.GetOptions{})
-	if err == nil {
-		return cd.Spec.Installed, nil
+	if err != nil {
+		return false, err
+	}
+
+	if cd.Spec.Installed {
+		return true, nil
+	}
+
+	checkFailureConditions := map[hivev1.ClusterDeploymentConditionType]corev1.ConditionStatus{
+		hivev1.ProvisionFailedCondition: corev1.ConditionTrue,
 	}
-	return false, err
+
+	for _, cond := range cd.Status.Conditions {
+		conditionStatus, found := checkFailureConditions[cond.Type]
+		if found && conditionStatus == cond.Status {
+			return false, fmt.Errorf("clusterdeployment has failed: %s == %s", cond.Type, cond.Status)
+		}
+	}
+
+	return false, nil
```

Comment:
wdyt about adding manager tests? I don't think we have any, and the logic here is growing

Reply:
Good point. I was racing to get this across the line, but there actually is time. I've added some tests.